jax/_src/numpy/fft.py

```diff
@@ -23,16 +23,29 @@
 
 
 def _fft_core(func_name, fft_type, a, s, axes, norm):
-  # TODO(skye): implement padding/cropping based on 's'.
   full_name = "jax.numpy.fft." + func_name
-  if s is not None:
-    raise NotImplementedError("%s only supports s=None, got %s" % (full_name, s))
+
+  if s is not None and not isinstance(s, list):
+    s = [s]
   if norm is not None:
     raise NotImplementedError("%s only supports norm=None, got %s" % (full_name, norm))
   if s is not None and axes is not None and len(s) != len(axes):
     # Same error as numpy.
     raise ValueError("Shape and axes have different lengths.")
 
+  if s is not None:
+    in_s = [x for x in s]
+    if fft_type == xla_client.FftType.IRFFT:
+      in_s[-1] = (in_s[-1] // 2 + 1)
+
+    if any([x != y for x, y in zip(a.shape, in_s)]):
+      # Cropping
+      a = a[tuple([slice(0, min(*x)) for x in zip(a.shape, in_s)])]
+      # Padding
+      a = jnp.asarray(a)
+      zeros = jnp.zeros(in_s, dtype=a.dtype)
+      a = jaxops.index_update(zeros, tuple([slice(0, x) for x in a.shape]), a)
```

Comment:
more succinct version of these three lines:
```
a = jnp.zeros(in_s, a.dtype).at[tuple(map(slice, a.shape))].set(a)
```

Reply:
There's also `jnp.pad`. Functionally I would guess they are very similar, though.

Reply:
I'd probably recommend using `lax.pad` or `jnp.pad` here, given I'm not completely confident XLA knows how to optimize a indexed-update of zeros into a `pad`, and we certainly prefer a `pad`.